src/v/cluster/rm_stm.cc

```diff
@@ -1217,7 +1243,9 @@ ss::future<> rm_stm::do_abort_old_txes() {
     for (auto pid : pids) {
         auto expiration_it = _mem_state.expiration.find(pid);
         if (expiration_it != _mem_state.expiration.end()) {
-            if (expiration_it->second.deadline() > clock_type::now()) {
+            if (
+              !expiration_it->second.is_expiration_requested
```

Comment:
Maybe it should be method in `expiration_info`
Smth like `is_expired`

Reply:
updated